src/lib/staged_ledger/check_commands.ml

```diff
@@ -24,7 +24,8 @@ let verify_command_with_transaction_pool_proxy
   let cmd_hash =
     (* PERF: `hash_command` is slow, so we may need to investigate if we could
        reuse hashes from transition verification. *)
-    User_command.of_verifiable verifiable_cmd |> Transaction_hash.hash_command
+    User_command.of_verifiable verifiable_cmd
+    |> User_command.read_all_proofs_from_disk |> Transaction_hash.hash_command
```

Comment:
Surely we don't need to read all of the proofs from disk for this. Is there a way to add a `hash_command_with_proofs_on_disk` version?

Reply:
will be addressed in a follow-up PR